File: problems/数学与位运算/136.只出现一次的数字.py
#!/usr/bin/env python3
"""
136. 只出现一次的数字
"""
def single_number(nums: list[int]) -> int:
    result = 0
    for num in nums:
        result ^= num
    return result

if __name__ == "__main__":
    # 测试用例
    print(single_number([2, 2, 1]))  # 预期: 1
    print(single_number([4, 1, 2, 1, 2]))  # 预期: 4
    print(single_number([1]))  # 预期: 1

File: problems/数学与位运算/136.只出现一次的数字.cpp
#include <iostream>
#include <vector>
using namespace std;

int singleNumber(vector<int>& nums) {
    int result = 0;
    for (int num : nums) result ^= num;
    return result;
}

int main() {
    // 测试用例
    vector<int> nums1 = {2, 2, 1};
    cout << singleNumber(nums1) << endl;  // 预期: 1
    vector<int> nums2 = {4, 1, 2, 1, 2};
    cout << singleNumber(nums2) << endl;  // 预期: 4
    vector<int> nums3 = {1};
    cout << singleNumber(nums3) << endl;  // 预期: 1
    return 0;
}
